### PR TITLE
Updating capture events + test.

### PR DESCRIPTION
**Pull request by dzearing**
Please fill in this template.
- [X] Prefer to make your PR against the `types-2.0` branch.
- [X] The package does not provide its own types, and you can not add them.
- [X] Test the change in your own code.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If adding a new definition:
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Run `tsc` without errors.
- [ ] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header.

If changing an existing definition:
- [X] Provide a URL to  documentation or source code which provides context for the suggested changes: https://facebook.github.io/react/docs/events.html#supported-events
- [ ] Increase the version number in the header if appropriate.

### CODE REVIEW

**Comment by vvakame** [comment]
*react/index.d.ts*

to authors (@pspeter3 @vsiao @johnnyreilly AssureSign (account can't be detected) Microsoft (account can't be detected)). Could you review this PR?
:+1: or :-1:?

Checklist

* [ ] pass the Travis CI test?

File: react/index.d.ts [hunk 1]
```diff
@@ -528,6 +528,101 @@ declare namespace React {
 
         // Transition Events
         onTransitionEnd?: TransitionEventHandler;
+
+        // Clipboard Events (captured)
+        onCopyCapture?: ClipboardEventHandler<T>;
```
Comment (vsiao) [inline react/index.d.ts]: Can you insert these inline instead? eg.

```ts
onCopy?: ClipboardEventHandler<T>;
onCopyCapture?: ClipboardEventHandler<T>;
onCut?: ClipboardEventHandler<T>;
onCutCapture?: ClipboardEventHandler<T>;
```

Thanks!

Comment (dzearing) [inline react/index.d.ts]: Absolutely!

Comment (dzearing) [inline react/index.d.ts]: Done.

File: react/index.d.ts [hunk 2]
```diff
@@ -436,98 +436,165 @@ declare namespace React {
...
         onDragStart?: DragEventHandler<T>;
+        onDragStartCapture?: DragEventHandler<T>;
         onDrop?: DragEventHandler<T>;
+        onDropCapture?: DragEventHandler<T>;
         onMouseDown?: MouseEventHandler<T>;
+        onMouseDownCapture?: MouseEventHandler<T>;
         onMouseEnter?: MouseEventHandler<T>;
+        onMouseEnterCapture?: MouseEventHandler<T>;
         onMouseLeave?: MouseEventHandler<T>;
+        onMouseLeaveCapture?: MouseEventHandler<T>;
```
Comment (vsiao) [inline react/index.d.ts]: `mouseenter` and `mouseleave` events do not have a capture phase. otherwise lgtm!

Comment (dzearing) [inline react/index.d.ts]: Gotcha. Removed. Good catch.

**Comment by mohsen1** [comment]
LGTM 

Please add mousewheel events too

```ts
        onMouseWheel?: MouseEventHandler<T>;
        onMouseWheelCapture?: MouseEventHandler<T>;
```

**Comment by vvakame** [comment]
@mohsen1 please send a new pull request!

**Pull request merged**